### PR TITLE
Pies are now refactored for new foods.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

Moves over pies to the newfood typepaths, as well as the few select pie slices.
I'm not sure if I followed the behavior with bonus reagents correctly so I'll defer to @Qustinnus if I need to re-do the reagents once more because he wasn't active at that time.

## Why It's Good For The Game

Please god, just finish the refactor floyd, I can't re-add food exports until it's done.

## Changelog
:cl:
refactor: Pies use the new food refactor and processing component.
/:cl:

### CODE REVIEW

File: code/game/objects/items/food/pie.dm [hunk 1]
```diff
@@ -0,0 +1,267 @@
+
```
Comment (Jared-Fogle) [inline code/game/objects/items/food/pie.dm:1]:
```suggestion
```

File: code/game/objects/items/food/pie.dm [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+/obj/item/food/pie/cream/proc/splat(atom/movable/hit_atom)
+	if(isliving(loc)) //someone caught us!
+		return
+	var/turf/T = get_turf(hit_atom)
+	new/obj/effect/decal/cleanable/food/pie_smudge(T)
+	if(reagents?.total_volume)
+		reagents.expose(hit_atom, TOUCH)
+	if(isliving(hit_atom))
+		var/mob/living/L = hit_atom
```
Comment (Jared-Fogle) [inline code/game/objects/items/food/pie.dm:50]: Better variable names

File: code/game/objects/items/food/pie.dm [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+	desc = "I bet you love stuff made out of plump helmets!"
+	icon_state = "plump_pie"
+	food_reagents = list(/datum/reagent/consumable/nutriment = 11, /datum/reagent/consumable/nutriment/vitamin = 4)
+	tastes = list("pie" = 1, "mushroom" = 1)
+	foodtypes = GRAIN | VEGETABLES
+
+/obj/item/food/pie/plump_pie/Initialize()
+	. = ..()
+	var/fey = prob(10)
+	if(fey)
```
Comment (Qustinnus) [inline code/game/objects/items/food/pie.dm:115]: Initialize needs to call parent last in this case. Food reagents are added to the component on init. Also what is up with the two if(fey" checks and omnizine additions

**Comment by LemonInTheDark** [comment]
How sits things?

**Comment by ArcaneMusic** [comment]
All is well with this, pies work functionally and I made sure that the sizes of the refactored new_foods all fit as intended.

**Pull request merged**